appstudio-controller/controllers/appstudio.redhat.com/snapshotenvironmentbinding_controller_test.go

```diff
@@ -1177,6 +1179,186 @@ var _ = Describe("SnapshotEnvironmentBinding Reconciler Tests", func() {
 		})
 
 	})
+
+	Context("Unit tests of non-reconcile functions", func() {
+
+		ctx := context.Background()
+
+		var k8sClient sharedutil.ProxyClient
+		var apiNamespace corev1.Namespace
+		log := log.FromContext(ctx)
+
+		var eventReceiver sharedutil.ListEventReceiver
+
+		BeforeEach(func() {
+			scheme,
+				argocdNamespace,
+				kubesystemNamespace,
+				namespace,
+				err := tests.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			err = appstudiosharedv1.AddToScheme(scheme)
+			Expect(err).To(BeNil())
+
+			apiNamespace = *namespace
+
+			// Create fake client
+			fakeK8sClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(namespace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			k8sClient = sharedutil.ProxyClient{
+				InnerClient: fakeK8sClient,
+				Informer:    &eventReceiver,
+			}
+
+		})
+
+		DescribeTable("verify updateBindingConditionOfSEB works as expected",
+			func(preCondition []metav1.Condition, newCondition metav1.Condition, expectedResult []metav1.Condition, expectUpdateStatusCalled bool) {
+
+				By("creating an initial SEB with the pre- state of bindingConditions")
+				env := appstudiosharedv1.SnapshotEnvironmentBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "my-seb",
+						Namespace: apiNamespace.Name,
+					},
+				}
+
+				err := k8sClient.Create(ctx, &env)
+				Expect(err).To(BeNil())
+				env.Status.BindingConditions = preCondition
+				err = k8sClient.Update(ctx, &env)
+				Expect(err).To(BeNil())
+
+				By("reseting the list of events after we updated the binding conditions")
```

Comment:
resetting

Reply:
Ugh, the english language is the worst :stuck_out_tongue: